### PR TITLE
Added CSharpExpressionDescriber, ExpressionExtensions

### PR DESCRIPTION
**Pull request by shanielh**

### CODE REVIEW

**Comment by shanielh** [comment]
I'm going to merge this one as it's too big to merge later, I'll start an issue about things to consider with the new DLL.

File: src/GraphClimber/ExpressionCompiler/Extensions/ExpressionExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Linq.Expressions;
+
+namespace GraphClimber.ExpressionCompiler.Extensions
+{
+    /// <summary>
+    /// Provides extension methods to create
+    /// expressions easier.
+    /// </summary>
+    public static class ExpressionExtensions
```
Comment (darkl) [inline src/GraphClimber/ExpressionCompiler/Extensions/ExpressionExtensions.cs:10]: Make this class internal, as it spams the intellisense.

File: src/GraphClimber/ExpressionCompiler/Extensions/ExpressionExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        public static readonly ConstantExpression Null = Expression.Constant(null);
+
+        /// <summary>
+        /// Creates an <see cref="ConstantExpression"/>
+        /// with the given value
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="value"></param>
+        /// <returns></returns>
+        public static Expression Constant<T>(this T value)
```
Comment (darkl) [inline src/GraphClimber/ExpressionCompiler/Extensions/ExpressionExtensions.cs:25]: Or, perhaps, you can make this method internal. This one really spam the intellisense.

File: src/GraphClimber/ExpressionCompiler/Extensions/ExpressionExtensions.cs (hunk 1 above)
Comment (shanielh) [inline src/GraphClimber/ExpressionCompiler/Extensions/ExpressionExtensions.cs:10]: [Done](https://github.com/Code-Sharp/GraphClimber/commit/a5b5cffeb4718d3f26f1a95399d1d463cf11fcf3) .

**Pull request merged**